### ISSUE TITLE
CORS setup not working - Cross-Origin Resource Sharing

### ISSUE THREAD

**Issue by PedroReyes** [sintaxi/surge#478]
Hello,

## 👁 Context

I am trying to access a json file I have stored in my surge.sh website. The content of the website is:

- an HTML file with an empty `<body>` 
- the JSON file, 
- and a CORS file with the content `*`.

I simply use the next command `echo '*' > CORS`

## ❌ Problem

I have followed the next steps: https://surge.sh/help/enabling-cross-origin-resource-sharing

However, when I try to access the json file located here: https://obedient-beam.surge.sh/deployments. Using the next code:

```javascript
let result = await fetch("http://obedient-beam.surge.sh/deployments", {
    method: "GET",
    headers: new Headers({
      "Content-type": "application/json",
      Accept: "application/json",
    }),
    mode: "no-cors",
    cache: "default",
    credentials: "include",
});
console.log(result);
```

I get the next opaque empty response as if no CORS had been setup:
```
Response {type: 'opaque', url: '', redirected: false, status: 0, ok: false, …}
body: null
bodyUsed: false
headers: Headers {}
ok: false
redirected: false
status: 0
statusText: ""
type: "opaque"
url: ""
```

Thanks in advance for the help.

Best regards,
Pedro Reyes.

**Comment by sintaxi** [comment]
Hi Pedro, I'm curious why are you using `mode: 'no-cors'`? Doesn't that block content coming from other origins?

**Comment by PedroReyes** [comment]
Hello @sintaxi ,

If I use `mode: 'no-cors'` the result is:

```
Access to fetch at 'https://obedient-beam.surge.sh/deployments' 
(redirected from 'http://obedient-beam.surge.sh/deployments') 
from origin 'http://localhost:6006' has been blocked by CORS policy: 
Response to preflight request doesn't pass access control check: 
The value of the 'Access-Control-Allow-Origin' header in the response
 must not be the wildcard '*' when the request's credentials mode is 'include'
```

And if I try to remove the mode:

```
Access to fetch at 'https://obedient-beam.surge.sh/deployments' 
from origin 'http://localhost:6006' has been blocked by CORS policy: 
Response to preflight request doesn't pass access control check: 
It does not have HTTP ok status.
```

I don't know what I am doing wrong 🤔

**Comment by fu-sen** [comment]
By any chance, are you using it with Surge Free? Since this is a Surge Professional only feature.

https://surge.sh/help/enabling-cross-origin-resource-sharing
It's listed at the bottom of this page. I thought it could be at the top of the page.

**Comment by PedroReyes** [comment]
Hello @fu-sen ,

You are completely right, I missed that panel. To be honest I thought it was an ad so I ignored him completely from my reading.

![image](https://user-images.githubusercontent.com/4096108/188119231-18dc41d3-260e-40bd-be45-29ef9c60c540.png)

The same happened to me with the follow along - I found that repository from the wild internet, not from the docs 😅

![image](https://user-images.githubusercontent.com/4096108/188119879-238b9e1c-d780-42e1-8c79-19d9c91a8eed.png)

Thanks @fu-sen for helping me out.

Best regards,
Pedro Reyes.
